pkg/component/runtime/service.go

```diff
@@ -432,10 +432,43 @@ func (s *ServiceRuntime) uninstall(ctx context.Context) error {
 }
 
 // UninstallService uninstalls the service
-func UninstallService(ctx context.Context, log *logger.Logger, comp component.Component) error {
+func UninstallService(ctx context.Context, log *logger.Logger, comp component.Component, uninstallToken string) error {
+	if comp.InputSpec.Spec.Service.Operations.Uninstall != nil {
+		// Resolve uninstall token parameter
+		comp.InputSpec.Spec.Service.Operations.Uninstall = resolveUninstallTokenArg(comp.InputSpec.Spec.Service.Operations.Uninstall, uninstallToken)
+	}
 	return uninstallService(ctx, log, comp, executeServiceCommand)
 }
 
+const uninstallTokenArg = "--uninstall-token"
```

Comment:
I was not sure if there is a better way to accomplish this with existing Agent implementation.
With this implementation the code looks for --uninstall-token arg in the uninstall command spec. And if there is that argument specified in the spec then the token value is getting injected into the args if the token value not empty string. If the token value is empty then the --uninstall-token arg is removed completely from the args.